src/Consolonia.NUnit/TestHelpers.cs

```diff
@@ -7,43 +8,90 @@ namespace Consolonia.NUnit
+        /// <summary>
+        /// Assert text pattern(s) is present in the console buffer.
+        /// Patterns starting and ending with '/' are treated as regex (e.g., /pattern/).
+        /// </summary>
+        /// <param name="unitTestConsole"></param>
+        /// <param name="patterns">Patterns to search for. Use /pattern/ syntax for regex.</param>
+        /// <returns></returns>
+        public static async Task AssertHasText(this UnitTestConsole unitTestConsole, params string[] patterns)
         {
-            foreach (string regexString in regexesToSearch)
-                await unitTestConsole.AssertHasText(regexString).ConfigureAwait(true);
-        }
-
-        public static async Task AssertHasText(this UnitTestConsole unitTestConsole, string regexToSearch)
-        {
-            (bool found, string bufferText) = await unitTestConsole.HasText(regexToSearch).ConfigureAwait(true);
+            await Dispatcher.UIThread.InvokeAsync(() =>
+            {
+                string printBuffer = unitTestConsole.PixelBuffer.PrintBuffer();
 
-            Assert.IsTrue(found,
-                $"'{regexToSearch}' was not found at the buffer: \r\n" + bufferText);
+                foreach (string pattern in patterns)
+                {
+                    if (IsRegexPattern(pattern))
+                    {
+                        var regex = new Regex(StripRegexDelimiters(pattern));
+                        bool found = regex.IsMatch(printBuffer);
+                        Assert.IsTrue(found,
+                            $"Regex '{pattern}' was not found at the buffer: \r\n" + printBuffer);
+                    }
+                    else
+                    {
+                        Assert.IsTrue(printBuffer.Contains(pattern, StringComparison.Ordinal), 
+                            $"'{pattern}' was not found at the buffer: \r\n" + printBuffer);
+                    }
+                }
+            });
         }
 
-        public static async Task AssertHasNoText(this UnitTestConsole unitTestConsole, string regexToSearch)
-        {
-            (bool found, string bufferText) = await unitTestConsole.HasText(regexToSearch).ConfigureAwait(true);
-
-            Assert.IsFalse(found,
-                $"'{regexToSearch}' was found at the buffer: \r\n" + bufferText);
-        }
 
-        private static async Task<(bool, string)> HasText(this UnitTestConsole unitTestConsole, string regexToSearch)
+        /// <summary>
+        /// Assert text pattern(s) is NOT present in the console buffer.
+        /// Patterns starting and ending with '/' are treated as regex (e.g., /pattern/).
+        /// </summary>
+        /// <param name="unitTestConsole"></param>
+        /// <param name="patterns">Patterns to search for. Use /pattern/ syntax for regex.</param>
+        /// <returns></returns>
+        public static async Task AssertHasNoText(this UnitTestConsole unitTestConsole, params string[] patterns)
```

Comment:
Now this methods repeat 18 of 20 lines. Could you extract that to something like was before?